client/my-sites/site-settings/controller.js

```diff
@@ -118,31 +115,53 @@ module.exports = {
 	},
 
 	deleteSite( context ) {
-		let site = sites.getSelectedSite();
+		const { dispatch, getState } = context.store;
+		const siteSlug = getSelectedSiteSlug( getState() );
+		const siteId = getSelectedSiteId( getState() );
 
-		if ( sites.initialized ) {
-			if ( ! canDeleteSite( site ) ) {
-				return page( '/settings/general/' + site.slug );
+		const redirectIfCantDelete = () => {
+			if ( ! canDeleteSite( getState(), getSelectedSite( getState() ) ) ) {
```

Comment:
Nitpick: how about storing `getState()` in a variable in order to call it only once? That applies for the rest of the usages throughout the controller.